### PR TITLE
[DOXIA-670] Drop dead XdocSink#link(String, String) method

### PR DESCRIPTION
**Pull request by michael-o**
https://issues.apache.org/jira/browse/DOXIA-670

### CODE REVIEW

**Comment by michael-o** [comment]
Note: This change requires to drop: https://github.com/apache/maven-doxia/blob/3b1499d0572718fa4dcb7c6b775bedbe757d46c5/doxia-modules/doxia-module-xdoc/src/main/resources/xdoc-2.0.xsd#L2923-L2935 and release a new version. Alternatively, we'd translate source to `<div class="source"><pre>...`

File: doxia-modules/doxia-module-xdoc/src/main/java/org/apache/maven/doxia/module/xdoc/XdocSink.java [hunk 1]
```diff
@@ -382,119 +377,6 @@ else if ( depth == SECTION_LEVEL_5 )
...
 
-    /**
-     * {@inheritDoc}
-     *
-     * @see XdocMarkup#SOURCE_TAG
-     * @see javax.swing.text.html.HTML.Tag#PRE
-     * @param attributes a {@link org.apache.maven.doxia.sink.SinkEventAttributes} object.
-     */
-    public void verbatim( SinkEventAttributes attributes )
-    {
```
Comment (michael-o) [inline doxia-modules/doxia-module-xdoc/src/main/java/org/apache/maven/doxia/module/xdoc/XdocSink.java:393]: I don't have a better idea how to make it proper, but it is definitively broken with XHML5.

Comment (michael-o) [reply doxia-modules/doxia-module-xdoc/src/main/java/org/apache/maven/doxia/module/xdoc/XdocSink.java:393]: I think this statement is nonsense and this method needs to be retained:
* Xdoc uses element of HTML5, but is not HTML5
* When XdocParser writes to a sink that sink will never see `<source>` anymore.

We have here a mix of custom elements and HTML5.

**Comment by michael-o** [comment]
This is now a minimalist version. I will leave the table as is.

File: doxia-modules/doxia-module-xdoc/src/main/java/org/apache/maven/doxia/module/xdoc/XdocSink.java [hunk 2]
```diff
@@ -505,31 +497,6 @@ public void close()
...
-        att.addAttribute( Attribute.HREF, HtmlTools.escapeHTML( name ) );
-
-        if ( target != null )
-        {
-            att.addAttribute( Attribute.TARGET, target );
-        }
-
-        writeStartTag( A, att );
-    }
-
```
Comment (hboutemy) [inline doxia-modules/doxia-module-xdoc/src/main/java/org/apache/maven/doxia/module/xdoc/XdocSink.java:532]: it seems that this link(String, String) method is more than "duplicate code": it's dead code that was introduced into Doxia 1.1 and was never called outside the unit test
then I'm all in to remove this dead code and update the associated unit test, with proper Jira issue renaming

on other changes like VALIGN, that are unrelated to this dead code removal, I'd prefer have a separate commit and review to better understand

and I don't understand why there is a discussion on "source", which seems to be completely an unrelated discussion

**Review by hboutemy — CHANGES_REQUESTED** [review]
let's rename this issue into "drop dead link(String, String) code" and have VALIGN removal in a separate issue, with discussion on source also separate

File: doxia-modules/doxia-module-xdoc/src/main/java/org/apache/maven/doxia/module/xdoc/XdocSink.java (hunk 2 above)
Comment (michael-o) [reply doxia-modules/doxia-module-xdoc/src/main/java/org/apache/maven/doxia/module/xdoc/XdocSink.java:531]: @hboutemy Done, completely reduced. Attributes are in another ticket. As for the `source`. HTML5 has now a source element which collides with Xdoc's source. That is why I decided not to fiddle with at the moment. Forget about it.

**Comment by michael-o** [comment]
@hboutemy PR redone.

**Review by hboutemy — APPROVED** [review]
thanks for the hard work

**Comment by jira-importer** [comment]
Resolve #826

**Pull request merged**